src/server/scripts/Spells/spell_rogue.cpp

```diff
@@ -355,80 +577,199 @@ class spell_rog_rupture : public SpellScriptLoader
         }
 };
 
+// 14161 - Ruthlessness
+class spell_rog_ruthlessness : public AuraScript
+{
+    PrepareAuraScript(spell_rog_ruthlessness);
+
+    void HandleProc(AuraEffect* aurEff, ProcEventInfo& procInfo)
+    {
+        Unit* target = GetTarget();
+        uint32 effAmount = aurEff->CalculateAmount(target);
+
+        Optional<int32> cost = GetFinishingMoveCPCost(procInfo.GetProcSpell());
+        if (cost.has_value() && roll_chance_i(effAmount * (*cost)))
+            target->ModifyPower(POWER_COMBO_POINTS, 1);
+    }
+
+    void Register() override
+    {
+        OnEffectProc += AuraEffectProcFn(spell_rog_ruthlessness::HandleProc, EFFECT_0, SPELL_AURA_DUMMY);
+    }
+};
+
+// 185438 - Shadow Strike
+class spell_rog_shadowstrike : public SpellScript
+{
+    PrepareSpellScript(spell_rog_shadowstrike);
+
+    SpellCastResult HandleCheckCast()
+    {
+        // Because the premeditation aura is removed when we're out of stealth,
+        // when we reach HandleEnergize the aura won't be there, even if it was when player launched the spell
+        _hasPremeditationAura = GetCaster()->HasAura(SPELL_ROGUE_PREMEDITATION_AURA);
+        return SPELL_FAILED_SUCCESS;
+    }
+
+    void HandleEnergize(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        if (_hasPremeditationAura)
+        {
+            if (caster->HasAura(SPELL_ROGUE_SLICE_AND_DICE))
+                if (Aura* premeditationPassive = caster->GetAura(SPELL_ROGUE_PREMEDITATION_PASSIVE))
+                    if (AuraEffect const* auraEff = premeditationPassive->GetEffect(EFFECT_1))
+                        SetHitDamage(GetHitDamage() + auraEff->GetAmount());
+
+            // Grant up to 10 seconds of slice and dice
```

Comment:
Requires #26945

According to my tests on retail (~20 tries) its always 10s instead of "up to" 10
```cpp
    CastSpellExtraArgs args;
    args.TriggerFlags = TRIGGERED_FULL_MASK;
    args.AddSpellMod(SPELLVALUE_DURATION, 10*IN_MILLISECONDS); 
    target->CastSpell(target, SPELL_ROGUE_SLICE_AND_DICE, args);
```